test/core/ecm_test.py

```diff
@@ -90,3 +90,98 @@ class ComponentB(MagicMock):
             }
         }
         self.assertEqual(self.entity.get_state(), expected_state)
+
+
+class TestEntityComponentManager(unittest.TestCase):
+
+    def setUp(self):
+        self.ecm = EntityComponentManager()
+
+    def test_create_entity(self):
+        entity = self.ecm.create_entity()
+        self.assertIsInstance(entity, Entity)
+        self.assertIsNotNone(entity._entity_id)  # Ensure entity ID is assigned
+
+    def test_get_entities(self):
+        entities = self.ecm.get_entities()
+        self.assertEqual(len(entities), 0)
+
+        entity1 = self.ecm.create_entity()
+        entity2 = self.ecm.create_entity()
+
+        entities = self.ecm.get_entities()
+        self.assertEqual(len(entities), 2)
+        self.assertIn(entity1._entity_id, entities)
+        self.assertIn(entity2._entity_id, entities)
+
+    def test_get_entity(self):
+        entity1 = self.ecm.create_entity()
+        entity2 = self.ecm.create_entity()
+
+        retrieved_entity1 = self.ecm.get_entity(entity1._entity_id)
+        retrieved_entity2 = self.ecm.get_entity(entity2._entity_id)
+
+        self.assertEqual(entity1, retrieved_entity1)
+        self.assertEqual(entity2, retrieved_entity2)
+
+    def test_get_entities_with_component(self):
+        component_type = MagicMock()
+        entity1 = self.ecm.create_entity()
+        entity2 = self.ecm.create_entity()
+        entity3 = self.ecm.create_entity()
+
+        entity1.get_components = MagicMock(return_value=[MagicMock()])
+        entity2.get_components = MagicMock(return_value=[])
+        entity3.get_components = MagicMock(return_value=[MagicMock()])
+        print(entity1.get_components())
+        print(entity2.get_components())
+        print(entity3.get_components())
+        entities_with_component = self.ecm.get_entities_with_component(component_type)
+        self.assertEqual(len(entities_with_component), 2)
+        self.assertIn(entity1, entities_with_component)
+        self.assertIn(entity3, entities_with_component)
+
+    def test_get_entities_of_type(self):
+        entity_type = "TestType"
+        different_entity_type = "DifferentType"
```

Comment:
Move this next to line 147 so it is close to where it is used.

Reply:
Done.